tools/bigquery-data-consolidator/src/list_datasets.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Python utility to list all datasets in a given project.
+
+# import libraries.
+# from google.cloud import bigquery
+
+# function definition
+def dataset_list(client):
+    """
+    Accepts the project id as the input,
```

Comment:
if i am not mistaken, this does not accept project id an input, rather the bigquery client